### DISCUSSION TITLE
Hook for more nuanced handling of ActivityPub post transformation?

### DISCUSSION THREAD

**Discussion by avdi** [Automattic/wordpress-activitypub#787] [Q&A] [answered]
I was thinking of tweaking my site so that longer "standard format" posts go to ActivityPub as an excerpt and a link, but shorter "status"-format posts are posted as the full content without a link. Is there a hook someone can point me to where I can implement this? Thanks!

**Comment by pfefferle** [comment, marked as answer]
The plugin has a transformer mechanism. You can hook your own transformer here: https://github.com/Automattic/wordpress-activitypub/blob/0be726983e9ed70fd61dba8ba9ae717145978673/includes/transformer/class-factory.php#L18

Or you can change the "Activity-Object-Type" settings to "WordPress Post-Format" to see if that already does the job!?!

**Comment by avdi** [comment]
Terrific, I'll start there!

**Comment by avdi** [comment]
FWIW it looks like the `activitypub_object_content_template` hook is sufficient for my needs. Here's what I wound up with:

```php
add_filter('activitypub_object_content_template', 'avdicodes_activitypub_content_template', 10, 2);
function avdicodes_activitypub_content_template($template, $object) {
    $post_format = get_post_format($object) ?: 'standard';
    if('standard' === $post_format) {
        return "[ap_excerpt]\n\n[ap_permalink type=\"html\"]\n\n[ap_hashtags]";
    } else {
        return "[ap_content]\n\n[ap_hashtags]";
    }
}
```
